packages/@aws-cdk/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -0,0 +1,289 @@
+ */
+export class AthenaStartQueryExecution extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: AthenaStartQueryExecutionProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, AthenaStartQueryExecution.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = this.createPolicyStatements();
+  }
+
+  private createPolicyStatements(): iam.PolicyStatement[] {
+    const policyStatements = [
+      new iam.PolicyStatement({
+        resources: [
+          cdk.Stack.of(this).formatArn({
+            service: 'athena',
+            resource: 'datacatalog',
+            resourceName: this.props.queryExecutionContext?.catalogName ? this.props.queryExecutionContext?.catalogName : 'AwsDataCatalog',
+          }),
+          cdk.Stack.of(this).formatArn({
+            service: 'athena',
+            resource: 'workgroup',
+            resourceName: this.props.workGroup ? this.props.workGroup : 'primary',
+          }),
+
+        ],
+        actions: ['athena:getDataCatalog', 'athena:startQueryExecution'],
+      }),
+    ];
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['s3:AbortMultipartUpload',
+          's3:CreateBucket',
+          's3:GetBucketLocation',
+          's3:GetObject',
+          's3:ListBucket',
+          's3:ListBucketMultipartUploads',
+          's3:ListMultipartUploadParts',
+          's3:PutObject'],
+        resources: [this.props.resultConfiguration?.outputLocation ? this.props.resultConfiguration?.outputLocation : '*'], // Need S3 location where data is stored https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+      }),
+    );
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['lakeformation:GetDataAccess'],
+        resources: [this.props.resultConfiguration?.outputLocation ? this.props.resultConfiguration?.outputLocation : '*'], // Workflow role permissions https://docs.aws.amazon.com/lake-formation/latest/dg/permissions-reference.html
+      }),
+    );
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['glue:BatchCreatePartition',
+          'glue:BatchDeletePartition',
+          'glue:BatchDeleteTable',
+          'glue:BatchGetPartition',
+          'glue:CreateDatabase',
+          'glue:CreatePartition',
+          'glue:CreateTable',
+          'glue:DeleteDatabase',
+          'glue:DeletePartition',
+          'glue:DeleteTable',
```

Comment:
does this mean that any glue database/table can be deleted? 
the permissions here seem really wide

Reply:
Refer to https://github.com/aws/aws-cdk/pull/11045#discussion_r512174712